### PR TITLE
Java column vector NVString timestamp conversion

### PR DESCRIPTION
**Pull request by rwlee**
Adds conversion when casting String column vectors to timestamp column vectors in Java.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -1409,9 +1409,43 @@ public ColumnVector asDate64() {
    * @return A new vector allocated on the GPU
    */
   public ColumnVector asTimestamp(TimeUnit unit) {
+    if (type == DType.STRING)
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: nit: we tend to have brackets around all blocks in the java code.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -1409,9 +1409,43 @@ public ColumnVector asDate64() {
...
+      return asTimestamp(unit, "%Y-%m-%dT%H:%M:%SZ");
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Could you please at least document the default timestamp format you are using in the javadocs.  What happens if the format does not match what is provided?

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 3]
```diff
@@ -1409,9 +1409,43 @@ public ColumnVector asDate64() {
...
+   * @param unit time unit to parse the timestamp into.
+   * @param format strptime format specifier string of the timestamp. Used to parse and convert
+   *               the timestamp with. Supports %Y,%y,%m,%d,%H,%I,%p,%M,%S,%f,%z format specifiers.
+   * @return A new ColumnVector containing the long representations of the timestamps in the
+   *         original column vector.
+   */
+  public ColumnVector asTimestamp(TimeUnit unit, String format) {
+    assert type == DType.STRING : "A column of type string is required " +
+                                  "for .timestampToLong() operation";
+    assert unit != TimeUnit.NONE : "Time unit may not be TimeUnit.NONE";
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Nit: in other places in CUDF a TImeUnit.NONE really means TimeUnit.MILLISECONDS and we have updated them accordingly.  This is fine, but it might nice to stay consistent.

Comment (rwlee) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: I think it's reasonable to expect a user to provide a real TimeUnit, but since we're supporting constructors without the unit being specified I guess we should allow the same when casting/converting.

Should we also be supporting an asTimestamp() cast, where unit is defaulted to TimeUnit.NONE?

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Like I said this is a nit, and it really is just about consistency.  If I were designing TimeUnit there wouldn't be a NONE, but as there is and other APIs in CUDF treat it as milliseconds, it would be nice to do the same here.  I would prefer if we have an `asTimestamp()` without a TimeUnit that it defaults to milliseconds instead of NONE. Simply because not all of the C code is consistent in how it treats NONE and I think it would be less likely to cause possible problems.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/*
+1427,8/*
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Is this change intended?

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 5]
```diff
@@ -1406,12 +1406,61 @@ public ColumnVector asDate64() {
...
+   *               for full parsing format specification and documentation.
+   * @return A new ColumnVector containing the long representations of the timestamps in the
+   *         original column vector.
+   */
+  public ColumnVector asTimestamp(TimeUnit unit, String format) {
+    assert type == DType.STRING : "A column of type string is required " +
+                                  "for .timestampToLong() operation";
+    assert format != null : "Format string may not be NULL";
+    if (unit == TimeUnit.NONE) {
+      unit == TimeUnit.MILLISECONDS;
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: This should be a singel `=` not `==`

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 3 above)
Comment (rwlee) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: added the default cast, I think the consistency makes things a lot more intuitive and less frustrating to use.
